app/assets/javascripts/components/teacher/teacher_settings_page.es6.jsx

```diff
@@ -0,0 +1,287 @@
+class TeacherSettingsPage extends UserSettings {
+
+  static get propTypes() {
+    return {
+      teacher: React.PropTypes.object.isRequired,
+    };
+  }
+
+  constructor(props) {
+    super(props);
+
+    let person = this.props.person || {};
+    person.availability = [];
+    person.stripe_routing_number = "*****";
+    person.stripe_account_number = "*****";
+    person.stripe_address_line1 = "*****";
+    person.stripe_address_city = "*****";
+    person.stripe_address_state = "*****";
+    person.stripe_address_postal_code = "*****";
+    person.stripe_account_holder_dob = "*****";
+    person.stripe_account_holder_type = "*****";
+    person.stripe_account_holder_name = "*****";
+    person.stripe_country = "*****";
+    person.stripe_ssn_last_4 = "*****";
+
+    this.state = {
+      addModalIsVisible: false,
+      removeModalIsVisible: false,
+      person: this.props.person,
+      id: this.props.id,
+    }
+  }
+
+  componentDidMount() {
+    this.fetchProfile();
+    this.fetchInstruments();
+  }
+
+  fetchProfile() {
+    let route = ApiConstants.teachers.show(this.props.id);
+
+    const resolve = (response) => {
+      this.props.fetchProfile();
+      const { person } = this.state;
+      Object.assign(person, response.teacher)
+
+      this.setState({ person: person });
+    }
+
+    const reject = (response) => console.log(response);
+    Requester.get(route, resolve, reject);
+  }
+
+  openRemoveModal() {
+    this.setState({ removeModalIsVisible: true });
+  }
+
+  closeRemoveModal() {
+    this.setState({ removeModalIsVisible: false });
+  }
+
+  openAddModal() {
+    this.setState({ addModalIsVisible: true });
+  }
+
+  closeAddModal() {
+    this.setState({ addModalIsVisible: false });
+  }
+
+  renderRemoveModal(instrument) {
+    const { removeModalIsVisible } = this.state;
+
+    if (removeModalIsVisible) {
+      return (
+        <RemoveInstrumentModal
+          isVisible={removeModalIsVisible}
+          handleClose={() => this.closeRemoveModal()}
+          fetchInstruments={() => this.fetchInstruments()}
```

Comment:
i think it would also work to use 
`handleClose={this.closeRemoveModal}` 
and `fetchInstruments={this.fetchInstruments}` 
to simplify it a bit